File: gin/util/base64.go
package util

import (
	"bytes"
	"compress/zlib"
	"encoding/base64"
	"io"
	"regexp"
)

type Base64 struct{}

/* 编码 */
func (Base64) Encode(data string) string {
	return base64.StdEncoding.EncodeToString([]byte(data))
}

/* 解码 */
func (Base64) Decode(data string) []byte {
	res, _ := base64.StdEncoding.DecodeString(data)
	return res
}

/* 编码(URL) */
func (Base64) UrlEncode(data []byte) string {
	// 编码
	res := base64.StdEncoding.EncodeToString(data)
	// 替换
	replace := map[string]string{"\\+": "*", "/": "-", "=": "_"}
	for k, v := range replace {
		reg, _ := regexp.Compile(k)
		res = reg.ReplaceAllString(res, v)
	}
	return res
}

/* 解码(URL) */
func (Base64) UrlDecode(data string) []byte {
	// 替换
	replace := map[string]string{"\\*": "+", "-": "/", "_": "="}
	for k, v := range replace {
		reg, _ := regexp.Compile(k)
		data = reg.ReplaceAllString(data, v)
	}
	// 解码
	res, _ := base64.StdEncoding.DecodeString(data)
	return res
}

/* 压缩 */
func (Base64) Compress(data []byte) []byte {
	var b bytes.Buffer
	w := zlib.NewWriter(&b)
	w.Write(data)
	w.Close()
	return b.Bytes()
}

/* 解压 */
func (Base64) UnCompress(data []byte) []byte {
	var b bytes.Buffer
	r, err := zlib.NewReader(bytes.NewReader(data))
	if err != nil {
		return nil
	}
	io.Copy(&b, r)
	r.Close()
	return b.Bytes()
}

File: phalcon/util/Base64.php
<?php
namespace Util;

class Base64 {

  /* 编码 */
  static function Encode(string $data): string {
    return base64_encode($data);
  }

  /* 解码 */
  static function Decode(string $data) {
    return base64_decode($data);
  }

  /* 编码(URL) */
  static function UrlEncode(string $data): string {
    // 编码
    $res = base64_encode($data);
    // 替换
    static $replace = Array('+'=>'*', '/'=>'-', '='=>'_');
    return str_replace(array_keys($replace), array_values($replace), $res);
  }

  /* 解码(URL) */
  static function UrlDecode(string $data) {
    // 替换
    static $replace = Array('+'=>'*', '/'=>'-', '='=>'_');
    $res = str_replace(array_values($replace), array_keys($replace), $data);
    // 解码
    return base64_decode($res);
  }

  /* 压缩 */
  static function Compress(string $data) {
    return gzcompress($data);
  }

  /* 解压 */
  static function UnCompress(string $data) {
    return gzuncompress($data);
  }

  /* 获取后缀 */
  static function GetExt(string $base64Type) {
    $ext = '';
    if($base64Type=='data:image/jpeg;base64') $ext='jpg';
    elseif($base64Type=='data:image/png;base64') $ext='png';
    elseif($base64Type=='data:image/gif;base64') $ext='gif';
    return $ext;
  }

}